### PR TITLE
Move the new AddResourceConfiguration to the Abstractions project

### PR DESCRIPTION
**Pull request by MikeAlhayek**

### CODE REVIEW

**Comment by Piedone** [comment]
I contemplated putting this there (together with the `.Core` project) but it's not actually an abstraction: it's neither an interface nor a fundamental concrete type, but a concrete implementation that needs to be like this only because `ResourceManager` uses `ResourceManagementOptions` like this. Solely from `ResourceManagementOptions` it doesn't follow that we should have this registration (and still, it's not an abstraction).

**Comment by MikeAlhayek** [comment]
I don't see an issue with placing it in the abstraction. The ResourceManagement is part of the abstraction, so to implement a new resource management, you'll need the abstractions only. So it does not make sense to include other project just to use that extension.  I think it's fair to put it in the abstraction project so people don't need to include yet another library. If we move it to core, they would have to switch from using Abstractions to Core.

File: src/OrchardCore/OrchardCore.ResourceManagement.Abstractions/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    /// <summary>
+    /// Adds a service implementing <see cref="IConfigureOptions{ResourceManagementOptions}"/> to the service collection.
+    /// </summary>
+    /// <typeparam name="T">
+    /// The type of the implementation of <see cref="IConfigureOptions{ResourceManagementOptions}"/> to register.
+    /// </typeparam>
+    public static IServiceCollection AddResourceConfiguration<T>(this IServiceCollection services)
+        where T : class, IConfigureOptions<ResourceManagementOptions>
+        => services.AddTransient<IConfigureOptions<ResourceManagementOptions>, T>();
+}
```
Comment (Piedone) [inline src/OrchardCore/OrchardCore.ResourceManagement.Abstractions/ServiceCollectionExtensions.cs:17]:
```suggestion
}

```

**Review by Piedone — APPROVED** [review]
OK, just be sure to revert the csproj changes under https://github.com/OrchardCMS/OrchardCore/pull/17095/files and feel free to merge.

**Pull request merged**